platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationEntry.java

```diff
@@ -0,0 +1,97 @@
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public default Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name) {
+        return getPropertyReferencedEntry(name, (r, n) -> true);
+    }
+
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified property
+     * in the properties file associated with this migration entry to be exported by the corresponding
+     * migratable.
+     * <p>
+     * The provided predicate will be invoked with the associated migration report and the property
+     * value (may be <code>null</code> if not defined). In case the property is not defined or is
+     * blank, an error will be automatically recorded unless the predicate returns <code>false</code>.
+     * Returning <code>true</code> in such case will still not create a corresponding migration entry.
+     * In all other cases, no errors or warning will be generated if the predicate returns <code>false</code>
+     * so it is up to the predicate to record one if required.
+     *
+     * @param name      the name of the property referencing a migration entry to create or retrieve
+     * @param validator a predicate to be invoked to validate the property value further which must
+     *                  return <code>true</code> to have a migration entry created
+     * @return a new migration entry or the existing one if already created for the referenced file
+     * from the specified property value or empty if the property is not defined, is blank or is invalid
+     * @throws IllegalArgumentException if <code>name</code> or <code>validator</code> is <code>null</code>
+     */
+    public Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name,
+            BiPredicate<MigrationReport, String> validator);
+
+    /**
+     * Stores this entry's content in the export using the specified exporter based on this entry's
+     * path which can include sub-directories.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report including
+     * those thrown by the exporter logic.
+     *
+     * @param exporter a consumer capable of exporting the content of this entry to a provided output stream
+     * @throws MigrationException       if a failure that prevents the operation from continue occurred
+     * @throws IllegalArgumentException if <code>exporter</code> is <code>null</code>
+     */
+    public void store(MigrationExporter exporter);
+
+    /**
+     * Gets an output stream for this entry which provides a low-level way for the migratable to
+     * store its own content in the export.
+     * <p>
+     * <i>Note:</i> The output stream will automatically be closed (if not closed already) when the
+     * output stream for another entry is retrieved or when calling {@link #store} on another entry.
```

Comment:
To me, a note meant to represent additional information worth knowing but either way if you ignore it, it won't create any problems as the implementation of these will take care of that for you.